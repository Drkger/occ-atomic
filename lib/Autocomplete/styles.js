import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
