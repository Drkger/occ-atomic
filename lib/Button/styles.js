import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};