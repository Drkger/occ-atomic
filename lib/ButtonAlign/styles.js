export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};