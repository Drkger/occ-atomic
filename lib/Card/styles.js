import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;