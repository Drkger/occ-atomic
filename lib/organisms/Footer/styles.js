import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';
import grid from '../../subatomic/grid';
import icons from '../../subatomic/icons';

export default {
    footer: {
        backgroundColor:colors.black,
        height:'25px',
        margin:'0',
        padding:'0 40px',
        fontSize:'12px',
        color:colors.grey2,
        fontFamily:fonts.title
    },
    footerList: {
        listStyle:'none',
        width:'auto',
        margin:'0',
        padding:'4px 0 0 0'
    },
    listElement: {
        listStyle:'none',
        float:'left',
        margin:'0 50px 0 0',
        padding:'0',
        position:'relative'
    },
    link: {
        color:colors.grey2,
        cursor:'pointer',
        textDecoration:'none',
        '&:hover, &:active, &:focus': {
            color:colors.white,
            textDecoration:'none'
        }
    },
    [`@media (max-width: ${grid.md-1}px)`]: {
        footer: {
            display: 'none !important'
        }
    },
    dropdownMenu: {
        display:'none',
        background:colors.black,
        position:'absolute',
        bottom:'calc(100% + 10px)',
        left:'50%',
        transform:'translateX(-50%)',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'12px',
        textAlign:'left',
        listStyle:'none',
        borderRadius:'4px',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        zIndex:'2'
    },
    dropdownElement: {
        float:'left',
        listStyle:'none'
    },
    dropdownLink: {
        color:colors.grey2,
        textDecoration:'none',
        padding:'3px 18px',
        lineHeight:'1.42857143',
        display:'block',
        '&:hover': {
            color:colors.white,
            textDecoration:'none'
        }
    }
};