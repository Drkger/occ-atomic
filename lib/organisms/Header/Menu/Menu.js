import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

class Menu extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
        this.toggleMenu = this.toggleMenu.bind(this);
        this.closeMenu = this.closeMenu.bind(this);
    }
    
    toggleMenu(e) {
        e.stopPropagation();
        this.setState({
            open: !this.state.open
        });
        if (!this.state.open) {
            window.addEventListener("click", this.closeMenu);
        }
    }
    
    closeMenu() {
        this.setState({
            open: false
        });
        window.removeEventListener("click", this.closeMenu);
    }
    
    render() {
        const { open } = this.state;
        const { classes, id, label, buttons } = this.props;
        return (
            <div id={id} className={classes.menu}>
                <div className={classes.btnContainer}>
                    <button className={`${classes.btnMenu} ${open ? classes.btnMenuOpen : ''}`} onClick={this.toggleMenu}>
                        <p className={classes.userName}>{label}</p>
                        <i className={classes.menuIcon} />
                    </button>
                    <ul className={`${classes.dropdownMenu} ${open ? classes.dropdownMenuOpen : ''}`}>
                        {buttons.map((button, index) => {
                            if (button == 'separator') {
                                return <li key={index} className={classes.separator} />;
                            } else {
                                return (<li key={button.id}>
                                    <a className={`${classes.listAnchor}${button.type == "signOut" ? ` ${classes.signOut}` : ''}`} {...button}>{button.label}</a>
                                </li>);
                            }
                        })}
                    </ul>
                </div>
            </div>
        );
    }
}

Menu.propTypes = {
    classes: PropTypes.object,
    config: PropTypes.object,
    label: PropTypes.string,
    buttons: PropTypes.array
};

export default Menu;