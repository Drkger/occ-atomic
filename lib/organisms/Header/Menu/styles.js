import colors from '../../../subatomic/colors';
import grid from '../../../subatomic/grid';
import icons from '../../../subatomic/icons';

export default {
    menu: {
        marginTop:'5px',
        display:'inline-block',
        cursor:'pointer !important',
    },
    btnContainer: {
        position:'relative',
        display:'inline-block',
        verticalAlign:'middle'
    },
    btnMenu: {
        cursor:'pointer',
        background:'none',
        border:'none',
        padding:'5px 10px 2px 10px',
        marginLeft:'10px',
        color:colors.white,
        transition:'0.3s all',
        outline:'none !important',
        borderRadius:'3px',
        '&:hover': {
            color:colors.blue,
            background:colors.white
        },
        '&:hover $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    btnMenuOpen: {
        color:colors.blue,
        background:colors.white,
        '& $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    userName: {
        paddingRight:'5px',
        margin:'0',
        display:'inline-block',
        overflow:'hidden',
        textOverflow:'ellipsis',
        maxWidth:'65px',
        whiteSpace:'nowrap'
    },
    menuIcon: {
        width:icons.menu.width,
        height:icons.menu.height,
        display:icons.menu.display,
        transition:'0.3s all',
        background:icons.base(icons.menu.icon(["#FFFFFF"]))
    },
    dropdownMenu: {
        position:'absolute',
        top:'100%',
        zIndex:'1000',
        display:'none',
        float:'left',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'14px',
        textAlign:'left',
        listStyle:'none',
        backgroundColor:colors.white,
        backgroundClip:'padding-box',
        border:'1px solid rgba(0,0,0,.15)',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        margin:'13px 0 0 -187px',
        right:'0'
    },
    dropdownMenuOpen: {
        display:'block'
    },
    listAnchor: {
        textDecoration:'none',
        display:'block',
        padding:'0 10px',
        margin:'5px 0 5px 0',
        clear:'both',
        fontWeight:'400',
        lineHeight:'1.42857143',
        whiteSpace:'nowrap',
        color:'#000',
        cursor:'pointer',
        '&:hover, &:focus': {
            textDecoration:'none',
            color:colors.blue
        }
    },
    signOut: {
        color:`${colors.pink} !important`,
        margin:'10px 0 5px 0'
    },
    separator: {
        height: '1px',
        background:'#ebebeb'
    },
    [`@media (max-width: ${grid.sm-1}px)`]: {
        dropdownMenu: {
            width:'200px'
        }
    }
};