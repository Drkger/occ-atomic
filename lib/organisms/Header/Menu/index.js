import injectSheet from 'react-jss';
import Menu from './Menu';
import styles from './styles';

export default injectSheet(styles)(Menu);