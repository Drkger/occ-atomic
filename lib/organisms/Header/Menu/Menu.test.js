import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';
import styles from './styles';
import MenuJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Menu", () => {

    it('matches the snapshot', () => {
        const buttons = [
            {
                id: 'item1',
                href:'http://occ.com.mx',
                label:'Item 1'
            }
        ];
        const wrapper = shallow(
            <Menu classes={classes} buttons={buttons} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("MenuJSS", () => {

    it('matches the snapshot', () => {
        const buttons = [
            {
                id: 'item1',
                href:'http://occ.com.mx',
                label:'Item 1'
            }
        ];
        const wrapper = shallow(
            <MenuJSS buttons={buttons} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});