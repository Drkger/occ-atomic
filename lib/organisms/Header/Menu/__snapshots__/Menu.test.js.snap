// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    buttons={
      Array [
        Object {
          "href": "http://occ.com.mx",
          "id": "item1",
          "label": "Item 1",
        },
      ]
    }
    classes={
      Object {
        "@media (max-width: 767px)": "@media (max-width: 767px)",
        "btnContainer": "btnContainer",
        "btnMenu": "btnMenu",
        "btnMenuOpen": "btnMenuOpen",
        "dropdownMenu": "dropdownMenu",
        "dropdownMenuOpen": "dropdownMenuOpen",
        "listAnchor": "listAnchor",
        "menu": "menu",
        "menuIcon": "menuIcon",
        "separator": "separator",
        "signOut": "signOut",
        "userName": "userName",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="btnContainer"
      >
        <button
          className="btnMenu "
          onClick={[Function]}
        >
          <p
            className="userName"
          />
          <i
            className="menuIcon"
          />
        </button>
        <ul
          className="dropdownMenu "
        >
          <li>
            <a
              className="listAnchor"
              href="http://occ.com.mx"
              id="item1"
              label="Item 1"
            >
              Item 1
            </a>
          </li>
        </ul>
      </div>,
      "className": "menu",
      "id": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btnMenu "
            onClick={[Function]}
          >
            <p
              className="userName"
            />
            <i
              className="menuIcon"
            />
          </button>,
          <ul
            className="dropdownMenu "
          >
            <li>
              <a
                className="listAnchor"
                href="http://occ.com.mx"
                id="item1"
                label="Item 1"
              >
                Item 1
              </a>
            </li>
          </ul>,
        ],
        "className": "btnContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="userName"
              />,
              <i
                className="menuIcon"
              />,
            ],
            "className": "btnMenu ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "userName",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "menuIcon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a
                  className="listAnchor"
                  href="http://occ.com.mx"
                  id="item1"
                  label="Item 1"
                >
                  Item 1
                </a>
              </li>,
            ],
            "className": "dropdownMenu ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "item1",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="listAnchor"
                  href="http://occ.com.mx"
                  id="item1"
                  label="Item 1"
                >
                  Item 1
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item 1",
                  "className": "listAnchor",
                  "href": "http://occ.com.mx",
                  "id": "item1",
                  "label": "Item 1",
                },
                "ref": null,
                "rendered": "Item 1",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="btnContainer"
        >
          <button
            className="btnMenu "
            onClick={[Function]}
          >
            <p
              className="userName"
            />
            <i
              className="menuIcon"
            />
          </button>
          <ul
            className="dropdownMenu "
          >
            <li>
              <a
                className="listAnchor"
                href="http://occ.com.mx"
                id="item1"
                label="Item 1"
              >
                Item 1
              </a>
            </li>
          </ul>
        </div>,
        "className": "menu",
        "id": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btnMenu "
              onClick={[Function]}
            >
              <p
                className="userName"
              />
              <i
                className="menuIcon"
              />
            </button>,
            <ul
              className="dropdownMenu "
            >
              <li>
                <a
                  className="listAnchor"
                  href="http://occ.com.mx"
                  id="item1"
                  label="Item 1"
                >
                  Item 1
                </a>
              </li>
            </ul>,
          ],
          "className": "btnContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="userName"
                />,
                <i
                  className="menuIcon"
                />,
              ],
              "className": "btnMenu ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "userName",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "menuIcon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <a
                    className="listAnchor"
                    href="http://occ.com.mx"
                    id="item1"
                    label="Item 1"
                  >
                    Item 1
                  </a>
                </li>,
              ],
              "className": "dropdownMenu ",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "item1",
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="listAnchor"
                    href="http://occ.com.mx"
                    id="item1"
                    label="Item 1"
                  >
                    Item 1
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Item 1",
                    "className": "listAnchor",
                    "href": "http://occ.com.mx",
                    "id": "item1",
                    "label": "Item 1",
                  },
                  "ref": null,
                  "rendered": "Item 1",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Menu styles matches the snapshot 1`] = `
Object {
  "@media (max-width: 767px)": Object {
    "dropdownMenu": Object {
      "width": "200px",
    },
  },
  "btnContainer": Object {
    "display": "inline-block",
    "position": "relative",
    "verticalAlign": "middle",
  },
  "btnMenu": Object {
    "&:hover": Object {
      "background": "#ffffff",
      "color": "#1476fb",
    },
    "&:hover $menuIcon": Object {
      "background": "url(data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgogICAgICAgICAgICAJIHZpZXdCb3g9IjAgMCAzLjEgMTMuMSIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMy4xIDEzLjE7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjMTQ3NmZiIiBjeD0iMS41IiBjeT0iMS41IiByPSIxLjUiLz4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjMTQ3NmZiIiBjeD0iMS41IiBjeT0iMTEuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDxjaXJjbGUgZmlsbD0iIzE0NzZmYiIgY3g9IjEuNSIgY3k9IjYuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDwvc3ZnPg==) no-repeat center center",
    },
    "background": "none",
    "border": "none",
    "borderRadius": "3px",
    "color": "#ffffff",
    "cursor": "pointer",
    "marginLeft": "10px",
    "outline": "none !important",
    "padding": "5px 10px 2px 10px",
    "transition": "0.3s all",
  },
  "btnMenuOpen": Object {
    "& $menuIcon": Object {
      "background": "url(data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgogICAgICAgICAgICAJIHZpZXdCb3g9IjAgMCAzLjEgMTMuMSIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMy4xIDEzLjE7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjMTQ3NmZiIiBjeD0iMS41IiBjeT0iMS41IiByPSIxLjUiLz4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjMTQ3NmZiIiBjeD0iMS41IiBjeT0iMTEuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDxjaXJjbGUgZmlsbD0iIzE0NzZmYiIgY3g9IjEuNSIgY3k9IjYuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDwvc3ZnPg==) no-repeat center center",
    },
    "background": "#ffffff",
    "color": "#1476fb",
  },
  "dropdownMenu": Object {
    "backgroundClip": "padding-box",
    "backgroundColor": "#ffffff",
    "border": "1px solid rgba(0,0,0,.15)",
    "boxShadow": "0 6px 12px rgba(0,0,0,.175)",
    "display": "none",
    "float": "left",
    "fontSize": "14px",
    "listStyle": "none",
    "margin": "13px 0 0 -187px",
    "minWidth": "160px",
    "padding": "5px 0",
    "position": "absolute",
    "right": "0",
    "textAlign": "left",
    "top": "100%",
    "zIndex": "1000",
  },
  "dropdownMenuOpen": Object {
    "display": "block",
  },
  "listAnchor": Object {
    "&:hover, &:focus": Object {
      "color": "#1476fb",
      "textDecoration": "none",
    },
    "clear": "both",
    "color": "#000",
    "cursor": "pointer",
    "display": "block",
    "fontWeight": "400",
    "lineHeight": "1.42857143",
    "margin": "5px 0 5px 0",
    "padding": "0 10px",
    "textDecoration": "none",
    "whiteSpace": "nowrap",
  },
  "menu": Object {
    "cursor": "pointer !important",
    "display": "inline-block",
    "marginTop": "5px",
  },
  "menuIcon": Object {
    "background": "url(data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgogICAgICAgICAgICAJIHZpZXdCb3g9IjAgMCAzLjEgMTMuMSIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMy4xIDEzLjE7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjRkZGRkZGIiBjeD0iMS41IiBjeT0iMS41IiByPSIxLjUiLz4KICAgICAgICAgICAgPGNpcmNsZSBmaWxsPSIjRkZGRkZGIiBjeD0iMS41IiBjeT0iMTEuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDxjaXJjbGUgZmlsbD0iI0ZGRkZGRiIgY3g9IjEuNSIgY3k9IjYuNSIgcj0iMS41Ii8+CiAgICAgICAgICAgIDwvc3ZnPg==) no-repeat center center",
    "display": "inline-block",
    "height": "16px",
    "transition": "0.3s all",
    "width": "16px",
  },
  "separator": Object {
    "background": "#ebebeb",
    "height": "1px",
  },
  "signOut": Object {
    "color": "#ff456a !important",
    "margin": "10px 0 5px 0",
  },
  "userName": Object {
    "display": "inline-block",
    "margin": "0",
    "maxWidth": "65px",
    "overflow": "hidden",
    "paddingRight": "5px",
    "textOverflow": "ellipsis",
    "whiteSpace": "nowrap",
  },
}
`;

exports[`MenuJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Menu)
    buttons={
      Array [
        Object {
          "href": "http://occ.com.mx",
          "id": "item1",
          "label": "Item 1",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        Object {
          "href": "http://occ.com.mx",
          "id": "item1",
          "label": "Item 1",
        },
      ],
      "classes": Object {
        "btnContainer": "Menu-btnContainer-0-1-2",
        "btnMenu": "Menu-btnMenu-0-1-3",
        "btnMenuOpen": "Menu-btnMenuOpen-0-1-4",
        "dropdownMenu": "Menu-dropdownMenu-0-1-7",
        "dropdownMenuOpen": "Menu-dropdownMenuOpen-0-1-8",
        "listAnchor": "Menu-listAnchor-0-1-9",
        "menu": "Menu-menu-0-1-1",
        "menuIcon": "Menu-menuIcon-0-1-6",
        "separator": "Menu-separator-0-1-11",
        "signOut": "Menu-signOut-0-1-10",
        "userName": "Menu-userName-0-1-5",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          Object {
            "href": "http://occ.com.mx",
            "id": "item1",
            "label": "Item 1",
          },
        ],
        "classes": Object {
          "btnContainer": "Menu-btnContainer-0-1-2",
          "btnMenu": "Menu-btnMenu-0-1-3",
          "btnMenuOpen": "Menu-btnMenuOpen-0-1-4",
          "dropdownMenu": "Menu-dropdownMenu-0-1-7",
          "dropdownMenuOpen": "Menu-dropdownMenuOpen-0-1-8",
          "listAnchor": "Menu-listAnchor-0-1-9",
          "menu": "Menu-menu-0-1-1",
          "menuIcon": "Menu-menuIcon-0-1-6",
          "separator": "Menu-separator-0-1-11",
          "signOut": "Menu-signOut-0-1-10",
          "userName": "Menu-userName-0-1-5",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
